src/lib.rs

```diff
@@ -48,11 +47,12 @@ impl linux_kernel_module::file_operations::Write for RandomFile {
         buf: &mut linux_kernel_module::user_ptr::UserSlicePtrReader,
         _offset: u64,
     ) -> linux_kernel_module::KernelResult<()> {
-        const TMP_BUF_LEN: usize = 256;
+        let mut chunkbuf = [0; 256];
         while !buf.is_empty() {
-            let mut tmp_buf = [0; TMP_BUF_LEN];
-            buf.read(&mut tmp_buf[..TMP_BUF_LEN.min(buf.len())])?;
-            linux_kernel_module::random::add_randomness(&tmp_buf[..TMP_BUF_LEN.min(buf.len())]);
+            let len = chunkbuf.len().min(buf.len());
+            let chunk = &mut chunkbuf[0..len];
+            buf.read(chunk);
```

Comment:
```suggestion
            buf.read(chunk)?;
```

Reply:
If I'd looked at rustc output I would have noticed this!